src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlObjectSerializerWriteContextComplex.cs

```diff
@@ -51,18 +51,31 @@ internal override bool WriteClrTypeInfo(XmlWriterDelegator xmlWriter, Type dataC
             return false;
         }
 
+#if NET_NATIVE
```

Comment:
As previous comment, some of these types/members are referenced in NetNative serialization component. We will have iterations to get rid of the directives and minimize these visibility differences.